### PR TITLE
Switch to EventSub API for stream online/offline and broadcaster follows

### PR DESCRIPTION
**Pull request by jwalter**
This should hopefully be a good start at least. I was only able to properly test the broadcaster follow events.

### CODE REVIEW

File: src/data/types.ts [hunk 1]
```diff
@@ -84,17 +84,12 @@ export interface VideoByUserIdResponse {
...
-  language: string;
   started_at: string;
-  tag_ids: string[];
-  thumbnail_url: string;
-  title: string;
   type: string;
-  user_id: string;
-  user_name: string;
-  viewer_count: number;
+  broadcaster_user_id: string;
```
Comment (whitep4nth3r) [inline src/data/types.ts:90]: Is the removal of all of these properties in this type going to mean we need to fetch the discord data for the announcements separately?

Comment (jwalter) [reply src/data/types.ts:90]: I believe so. Some of the data can be read using getLatestVideo.

Confession: I actually updated discord.js as well but that was during the first 2 hours when I was accidentally working on a stale local branch (that had other changes too) and then when I moved to changes manually to a fresh branch I seem to have forgotten the discord changes :(

Comment (whitep4nth3r) [reply src/data/types.ts:90]: the only problem is that when I go live... this might crash the bot :D

Comment (whitep4nth3r) [reply src/data/types.ts:90]: I'll take a look after I merge

**Pull request merged**